hawkbit-repository/hawkbit-repository-core/src/main/java/org/eclipse/hawkbit/event/BusProtoStuffMessageConverter.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.event;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.bus.event.RemoteApplicationEvent;
+import org.springframework.integration.support.MutableMessageHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.converter.AbstractMessageConverter;
+import org.springframework.messaging.converter.MessageConversionException;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+import org.springframework.util.MimeType;
+
+import io.protostuff.LinkedBuffer;
+import io.protostuff.ProtobufIOUtil;
+import io.protostuff.ProtostuffIOUtil;
+import io.protostuff.Schema;
+import io.protostuff.runtime.RuntimeSchema;
+
+/**
+ * A customize message converter for the spring cloud events. The converter is
+ * registered fot the application/binary+protostuff type.
+ *
+ */
+public class BusProtoStuffMessageConverter extends AbstractMessageConverter {
+
+    public static final MimeType APPLICATION_BINARY_PROTOSTUFF = new MimeType("application", "binary+protostuff");
+    private static final Logger LOG = LoggerFactory.getLogger(BusProtoStuffMessageConverter.class);
+    private static final String DEFAULT_CLASS_FIELD_NAME = "__Class__";
+
+    /**
+     * Constructor.
+     */
+    public BusProtoStuffMessageConverter() {
+        super(APPLICATION_BINARY_PROTOSTUFF);
+    }
+
+    @Override
+    protected boolean supports(final Class<?> aClass) {
+        return RemoteApplicationEvent.class.isAssignableFrom(aClass);
+    }
+
+    @Override
+    public Object convertFromInternal(final Message<?> message, final Class<?> targetClass,
+            final Object conversionHint) {
+        final Object payload = message.getPayload();
+
+        try {
+            final Class<?> eventClass = ClassUtils
+                    .getClass(message.getHeaders().get(DEFAULT_CLASS_FIELD_NAME).toString());
+            if (payload instanceof byte[]) {
```

Comment:
Is it possible/allowed that the payload is not of type `byte[]` and you'd return null?